torchrl/collectors/collectors.py

```diff
@@ -1266,9 +1270,28 @@ def _main_async_collector(
             # this is expected to happen if queue_out reached the timeout, but no new msg was waiting in the pipe
             # in that case, the main process probably expects the worker to continue collect data
             if has_timed_out:
+                counter = 0
+                # has_timed_out is True if the process failed to send data, which will
+                # typically occur if main has taken another batch (i.e. the queue is Full).
+                # In this case, msg is the previous msg sent by main, which will typically be "continue"
+                # If it's not the case, it is not expected that has_timed_out is True.
                 if msg not in ("continue", "continue_random"):
                     raise RuntimeError(f"Unexpected message after time out: msg={msg}")
             else:
+                # if has_timed_out is False, then the time out does not come from the fact that the queue is Full.
+                # this means that our process has been waiting for a command from main in vain, while main was not
+                # receiving data.
+                # This will occur if main is busy doing something else (e.g. computing loss etc).
+                counter += _timeout
+                if counter >= (_MAX_IDLE_COUNT * _TIMEOUT):
+                    raise RuntimeError(
+                        f"This process waited for {counter} seconds "
+                        f"without receiving a command from main. Consider increasing the maximum idle count "
+                        f"if this is expected via the environment variable MAX_IDLE_COUNT "
+                        f"(current value is {_MAX_IDLE_COUNT})."
+                        f"\nIf this occurs at the end of a function, it means that your collector has not been "
+                        f"collected, consider calling `collector.shutdown()` or `del collector` at the end of the function."
```

Comment:
I'm not sure that both `shutdown` and `del` will work. In my original code I added an error in the del function so it will be shown if the collector is not shutdown proparly.

Reply:
I will check that in the tests